pkg/builder.go

```diff
@@ -27,7 +27,7 @@
 	mu       sync.RWMutex
 	stop     bool
 	builder  *samplebuilder.SampleBuilder
-	elements map[string]Element
+	elements sync.Map
```

Comment:
if we already have a mutex, can we use that instead to synchronize access to the map?